app/helpers/patient_helper.rb

```diff
@@ -2,6 +2,209 @@
+                            url: 'ombCategory',
+                            valueCoding: FHIR::Coding.new(code: '2135-2', system: 'urn:oid:2.16.840.1.113883.6.238', display: 'Hispanic or Latino')
+                          ) : nil,
+                          ethnicity == 'Not Hispanic or Latino' ? FHIR::Extension.new(
+                            url: 'ombCategory',
+                            valueCoding: FHIR::Coding.new(code: '2186-5', system: 'urn:oid:2.16.840.1.113883.6.238', display: 'Not Hispanic or Latino')
+                          ) : nil,
+                          FHIR::Extension.new(
+                            url: 'text',
+                            valueString: ethnicity
+                          )
+                        ])
+  end
+
+  # Return a string representing the ethnicity of the given FHIR::Patient
+  def self.ethnicity(patient)
+    url = 'us-core-ethnicity'
+    code = patient&.extension&.select { |e| e.url.include?(url) }&.first&.extension&.select { |e| e.url == 'ombCategory' }&.first&.valueCoding&.code
+    return 'Hispanic or Latino' if code == '2135-2'
+    return 'Not Hispanic or Latino' if code == '2186-5'
+
+    nil
+  end
+
+  # Build a FHIR US Core BirthSex Extension given Sara Alert sex information.
+  def us_core_birthsex(sex)
+    # Don't return an extension if no sex specified
+    return nil unless %w[Male Female Unknown].include?(sex)
+
+    # Build out extension based on what sex was specified
+    code = sex == 'Unknown' ? 'UNK' : sex.first
+    FHIR::Extension.new(url: 'http://hl7.org/fhir/us/core/StructureDefinition/us-core-birthsex', valueCode: code)
+  end
+
+  # Return a string representing the birthsex of the given FHIR::Patient
+  def self.birthsex(patient)
+    url = 'us-core-birthsex'
+    code = patient&.extension&.select { |e| e.url.include?(url) }&.first&.valueCode
+    return 'Male' if code == 'M'
+    return 'Female' if code == 'F'
+    return 'Unknown' if code == 'UNK'
+
+    nil
+  end
+
+  # Helper to create an extension for preferred contact method
+  def to_preferred_contact_method_extension(preferred_contact_method)
+    preferred_contact_method.nil? ? nil : FHIR::Extension.new(
+      url: 'http://saraalert.org/preferred_contact_method',
+      valueString: preferred_contact_method
+    )
+  end
+
+  # Helper to understand an extension for preferred contact method
+  def self.from_preferred_contact_method_extension(patient)
+    pcm = patient&.extension&.select { |e| e.url.include?('preferred_contact_method') }&.first&.valueString
+    pcm = nil unless ['E-mailed Web Link', 'SMS Texted Weblink', 'Telephone call', 'SMS Text-message'].include?(pcm)
+    pcm
+  end
+
+  # Helper to create an extension for preferred contact time
+  def to_preferred_contact_time_extension(_preferred_contact_method)
+    preferred_contact_time.nil? ? nil : FHIR::Extension.new(
+      url: 'http://saraalert.org/preferred_contact_time',
```

Comment:
See previous comment about extension URLs. At some point we will want these to resolve, so something like http://saraalert.org/StructureDefinition/preferred_contact_time may be a better choice.